usaspending_api/common/logging.py

```diff
@@ -204,3 +222,84 @@ def ensure_logging(
             f"logger={cfg_logger_name}, formatter={cfg_formatter_name}, formatter_class={str(formatter_class)})"
         )
     return logger_to_use
+
+
+# CUSTOM Logging EXPORTER for debugging
+class LoggingSpanProcessor(SpanProcessor):
+    def on_end(self, span: ReadableSpan) -> None:
+        trace_id = span.context.trace_id
+        span_id = span.context.span_id
+        logger.debug(f"Span ended: trace_id={trace_id}, span_id={span_id}, {span.name}_attributes={span.attributes}")
+
+    def force_flush(self, timeout_millis: int = 30000) -> bool:
+        return True
+
+
+class CustomAttributeSpanProcessor(SpanProcessor):
+    def __init__(self, attribute_key, attribute_value):
+        self.attribute_key = attribute_key
+        self.attribute_value = attribute_value
+
+    def on_start(self, span, parent_context):
+        # Add the custom attribute when the span starts
+        span.set_attribute(self.attribute_key, self.attribute_value)
+
+    def on_end(self, span):
+        pass  # No action needed when the span ends
+
+
+def add_custom_attribute_span_processors(attribute_pairs):
+    """
+    Helper function to add custom attribute span processors to the tracer provider.
+
+    Args:
+        attribute_pairs (list of tuples): Each tuple contains (key, value) for the attribute.
```

Comment:
Not worth blocking over but you could move this up to the function definition as a type hint instead of defining here.

Reply:
Done.